modules/apps/27-interchain-accounts/controller/types/msgs.go

```diff
@@ -41,6 +42,10 @@ func (msg MsgRegisterInterchainAccount) ValidateBasic() error {
 		return errorsmod.Wrap(err, "invalid connection ID")
 	}
 
+	if !slices.Contains([]channeltypes.Order{channeltypes.ORDERED, channeltypes.UNORDERED}, msg.Order) {
```

Comment:
I asked about this in security audit and though it does indeed get handled later on I do think having explicit ordering check here also makes sense. Only worry is something like https://github.com/cosmos/ibc-go/pull/5668 occurring again, maybe we can add these checks in spec too?

Reply:
Sounds good to include it in the spec as well. I have [this PR](https://github.com/cosmos/ibc/pull/1047) with ICS 27 updates, so I can include it there.